crates/types/src/schema/mod.rs

```diff
@@ -31,16 +32,17 @@ use crate::net::metadata::MetadataContainer;
 use crate::net::metadata::MetadataKind;
 
 /// The schema information
-#[serde_as]
 #[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(from = "serde_hacks::Schema", into = "serde_hacks::Schema")]
```

Comment:
I think in other parts of the code we usually call them `SchemaShadow`. It would be good to unify things to reduce mental load.